addons/sale/tests/test_sale_report.py

```diff
@@ -17,66 +15,149 @@ class TestSaleReportCurrencyRate(SaleCommon):
+
+        # Create sale orders.
+        eur_cmp_ars_curr = self.env['sale.order'].with_company(self.eur_company).create({
             'partner_id': self.partner.id,
-            'pricelist_id': usd_pricelist.id,
+            'pricelist_id': ars_pricelist.id,
+            'date_order': fields.Date.today(),
             'order_line': [
                 Command.create({
                     'product_id': self.product.id,
                     'product_uom_qty': 1.0,
                 }),
             ]
         })
-        # the sales order's amount is in company currency
-        self.assertEqual(usd_order.amount_total, 20.0)
+        eur_cmp_usd_curr = self.env['sale.order'].with_company(self.eur_company).create({
+            'partner_id': self.partner.id,
+            'pricelist_id': usd_pricelist.id,
+            'date_order': fields.Date.today(),
+            'order_line': [
+                Command.create({
+                    'product_id': self.product.id,
+                    'product_uom_qty': 2.0,
+                }),
+            ]
+        })
+        eur_cmp_eur_curr = self.env['sale.order'].with_company(self.eur_company).create({
+            'partner_id': self.partner.id,
+            'pricelist_id': eur_pricelist.id,
+            'date_order': fields.Date.today(),
+            'order_line': [
+                Command.create({
+                    'product_id': self.product.id,
+                    'product_uom_qty': 3.0,
+                }),
+            ]
+        })
+        usd_cmp_usd_curr = self.env['sale.order'].with_company(self.usd_company).create({
+            'partner_id': self.partner.id,
+            'pricelist_id': usd_pricelist.id,
+            'date_order': fields.Date.today(),
+            'order_line': [
+                Command.create({
+                    'product_id': self.product.id,
+                    'product_uom_qty': 4.0,
+                }),
+            ]
+        })
+
+        self.assertEqual(eur_cmp_ars_curr.currency_rate, 20)
+        self.assertEqual(eur_cmp_usd_curr.currency_rate, 2)
+        self.assertEqual(eur_cmp_eur_curr.currency_rate, 1)
+        self.assertEqual(usd_cmp_usd_curr.currency_rate, 1)
+
+        # The report should show the amount in company currency
+        report_lines = self.env['sale.report'].sudo().with_context(allow_company_ids=[
+            self.usd_company.id,
+            self.eur_company.id,
+        ]).search(
+            [('order_id', 'in', (
+                eur_cmp_ars_curr.id,
+                eur_cmp_eur_curr.id,
+                eur_cmp_usd_curr.id,
+                usd_cmp_usd_curr.id,
+            ))])
 
-        # the report should match the amount on the SO
-        report_line = self.env['sale.report'].sudo().search(
-            [('order_id', '=', usd_order.id)])
-        self.assertEqual(report_line.price_total, 20.0)
+        price_total = sum(report_lines.mapped('price_total'))
+        self.assertEqual(price_total, 200.0)
```

Comment:
Maybe add a comment with the details on why you should get 200 as total. Either here or L156-159 which would also be clear